cni/pkg/ambient/informers_test.go

```diff
@@ -267,7 +264,7 @@ func (h *fakePodReconcileHandler) delPodFromMesh(pod *v1.Pod, event controllers.
 
 	h.events = append(h.events, podEvent{
 		event: "del",
-		pod:   pod,
+		pod:   pod.DeepCopy(),
```

Comment:
Instead of deepcopy here, i would suggest finding where it is mutated and prefer to fix it there.

Reply:
Agree with you, any idea to reproduce this kind of error in local. I tested it under the high cpu stress(99.9%), and also passed.

Reply:
go test -race -count

Reply:
It reproduced when set count to a big number.

Just realized the previous code has bug, it should return false in the forloop,

```go
assert.EventuallyEqual(t, func() bool {
	skippedEvents := 0
	for i, event := range c.expectedEvents {
           ...
           +++ return false +++
	}
	return true
}, true, retry.Timeout(3*time.Second))
```

This missed `return false` hided a frw problems.